kotlinx.fuzz.engine/src/main/kotlin/kotlinx/fuzz/KFuzzEngine.kt

```diff
@@ -20,6 +21,13 @@ interface KFuzzEngine {
      */
     fun runTarget(instance: Any, method: Method): Throwable?
 
+    /**
+     * Sets an instance to write reproducers
+     *
+     * @param reproducer
+     */
+    fun setReproducer(reproducer: CrashReproducerWriter)
```

Comment:
This feels a bit Javaesque. Why don't we declare it as a property?

Reply:
I didn't want it to be an abstract class, but fixed

Reply:
In Kotlin you can add properties to interfaces, you don't have to make them abstract classes:
```kotlin
interface Foo {
    var property: String
}
```